File: hackerrank/warmup/service-lane/service-lane.py
#!/usr/bin/env python

t = int(raw_input().split()[1]);

sizes = raw_input().split()

for i in xrange(t):
	input = raw_input().split();
	entry = int(input[0]);
	exit = int(input[1]) + 1;

	width = min(sizes[entry:exit]);

	print width

File: hackerrank/warmup/service-lane/service-lane.rb
#!/usr/bin/env ruby

t = gets.split[1].to_i

sizes = gets.split

(1..t).each { |i|
  input = gets.split
  entry = input[0].to_i
  exit = input[1].to_i

  path = sizes.slice(entry, exit - entry + 1)

  puts path.min
}
